src/Utils/AffineTransform.vala

```diff
@@ -36,14 +36,15 @@ public class Akira.Utils.AffineTransform : Object {
      */
     public static void move_from_event (
         CanvasItem item,
+        SnapManager.SnapMatchData data,
         double event_x,
         double event_y,
         ref double initial_event_x,
         ref double initial_event_y
     ) {
         // Calculate the delta between the initial point and new mouse location.
-        var delta_x = fix_size (event_x - initial_event_x);
-        var delta_y = fix_size (event_y - initial_event_y);
+        var delta_x = fix_size (event_x - initial_event_x + data.horizontal_data.snap_offset());
+        var delta_y = fix_size (event_y - initial_event_y + data.vertical_data.snap_offset());
```

Comment:
Here, to prevent the jitter we can do something like this:
```PHP
        var delta_x = fix_size (event_x - initial_event_x);
        var delta_y = fix_size (event_y - initial_event_y);

        if (
            (delta_x < 0 && delta_x >= -data.horizontal_data.snap_offset()) ||
            (delta_x > 0 && delta_x <= data.horizontal_data.snap_offset())
        ) {
            delta_x = 0;
        }
        if (
            (delta_y < 0 && delta_y >= -data.vertical_data.snap_offset()) ||
            (delta_y > 0 && delta_y <= data.vertical_data.snap_offset())
        ) {
            delta_y = 0;
        }
```

The thing that is missing here is to check if the actual matrix X and Y of the moved item match exactly the X and Y locations of the guides.
If they don't match, we should subtract or add the delta between those coordinates to the matrix to make the item snap in place.

What do you think?

Reply:
I think this is pretty close. I actually added some useful data in my current branch (need to push) to help with this. I got something very similar working:
```
        if (data.horizontal.type == SnapManager.MatchType.FUZZY) {
            delta_x = data.horizontal.snap_offset();
        }
        else if (data.horizontal.type == SnapManager.MatchType.EXACT) {
            delta_x = 0;
        }
```

This snaps correctly and stays in place...but the issue is that it never snaps out (black hole effect!). I need some kind of reference point to the original starting point of the drag--otherwise it's impossible to restore the correct position of the object relative to the mouse.

The downside to an approach like this, compared to my suggestion of a shadow of the selection, is that if we unsnap in a move tick, then an immediate snap would not trigger. It may not matter so much in practice...maybe? i.e., there is one frame where a potential snap may be missed.

Reply:
I added some information to make this work. I pushed the changes. I think working this relative to the previous positions will always be really flaky.

The snap calculation needs to be based on the theoretical position of the selection to allow the system to be deterministic.